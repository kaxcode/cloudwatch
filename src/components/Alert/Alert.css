.Alert {
  text-align: center;
  margin-top: 20px;
  background-color: white;
  width: 100%;
  height: 100%;
  z-index: 10000000;
  position: absolute;
}
