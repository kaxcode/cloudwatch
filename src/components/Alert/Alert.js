import { Component } from 'react';
import PropTypes from 'prop-types';

export default class Alert extends Component {
  componentDidMount = () => {
    if (this.props.show) {
      this.handleAlert();
    }
  };

  componentDidUpdate = prevProps => {
    if (this.props.show && this.props.show !== prevProps.show) {
      this.handleAlert();
    }
  };

  handleAlert = () => {
    alert(this.props.msg);
    this.props.onDismiss();
  };

  render() {
    return null;
  }
}

Alert.propTypes = {
  onDismiss: PropTypes.func.isRequired,
  msg: PropTypes.string,
  show: PropTypes.bool
};
