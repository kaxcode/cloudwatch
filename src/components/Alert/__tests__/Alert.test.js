import React from 'react';
import Enzyme from 'enzyme';
import Alert from '../Alert';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('Alert', () => {
  it('renders when show is true', () => {
    // Arrange
    let wrapper = mount(<Alert onDismiss={jest.fn()} show />);

    it('renders correctly', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('shows alert when start is clicked and time is zero', () => {
      expect(wrapper.find('Alert').props().show).toBe(true);
    });

    it('does not show alert when start is not clicked and time is zero', () => {
      wrapper = shallow(<Alert onDismiss={jest.fn()} show={false} />);
      expect(wrapper.find('Alert').props().show).toBe(false);
    });
  });
});
