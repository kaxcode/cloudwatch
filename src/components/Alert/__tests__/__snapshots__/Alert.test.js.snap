// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert does not render an Alert when :show is false 1`] = `""`;

exports[`Alert renders an Alert when :show is true 1`] = `
<button
  className="Alert"
  onClick={[Function]}
>
  <img
    alt="Time is up, click this to close alert"
    src="alert_cloud.svg"
  />
</button>
`;
